src/backend/src/utils/tasks.utils.ts

```diff
@@ -14,3 +15,93 @@ export const convertTaskStatus = (status: Task_Status): TaskStatus =>
     IN_PROGRESS: TaskStatus.IN_PROGRESS,
     DONE: TaskStatus.DONE
   }[status]);
+
+export const hasPermissionToEditTask = async (user: User, taskId: string): Promise<boolean> => {
+  if (user.role === Role.ADMIN || user.role === Role.APP_ADMIN) {
+    return true;
+  }
+
+  const task = await prisma.task.findUnique({
+    where: { taskId },
+    select: {
+      createdByUserId: true,
+      wbsElement: {
+        select: {
+          projectLeadId: true,
+          projectManagerId: true,
+          project: {
+            select: {
+              team: {
+                select: {
+                  leaderId: true,
+                  members: {
+                    select: {
+                      userId: true
+                    }
+                  }
+                }
+              }
+            }
+          },
+          workPackage: {
+            select: {
+              project: {
+                select: {
+                  team: {
+                    select: {
+                      leaderId: true,
+                      members: {
+                        select: {
+                          userId: true
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  });
+
+  if (!task) {
+    return false;
+  }
+
+  // Check if the user created the task
+  if (task.createdByUserId === user.userId) {
+    return true;
+  }
+
+  // Check if the task's wbsElement's projectLead or projectManager created the task
+  if (task.wbsElement.projectLeadId === user.userId) {
+    return true;
+  }
+  if (task.wbsElement.projectManagerId === user.userId) {
+    return true;
+  }
+
+  // Check if the user is the project leader or on the project team
+  if (task.wbsElement.project?.team?.leaderId === user.userId) {
+    return true;
+  }
+  const projectTeamUserIds = task?.wbsElement.project?.team?.members?.map((user) => user.userId);
```

Comment:
i dont think we should allow all members of the team to edit it, just the team lead